client/pages/routing.jsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { Map, GoogleApiWrapper, InfoWindow } from 'google-maps-react';
+
+class Routing extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onMapClicked = this.onMapClicked.bind(this);
+    this.makeRouteURL = this.makeRouteURL.bind(this);
+    this.state = {
+      vehicleLocation: {
+        lat: 40.854885,
+        lng: -88.081807
+      },
+      route: [],
+      counter: 1,
+      vehicle: {}
+    };
+
+    const vId = 2;
+    fetch(`/api/vehicleinfo/${vId}`)
+      .then(res => res.json())
+      .then(result => {
+        this.setState({ vehicle: result });
+      });
+
+  }
+
+  onMapClicked(mapProps, map, clickEvent) {
+
+    const eventRaw = JSON.stringify(clickEvent);
+    const event = JSON.parse(eventRaw);
```

Comment:
Is there a specific reason to stringify and then immediately parse the same data?